smart_open/smart_open_lib.py

```diff
@@ -635,12 +635,26 @@ def compression_wrapper(file_obj, filename, mode):
         return file_obj
 
 
-def file_smart_open(fname, mode='rb'):
+def file_smart_open(fname, mode='rb', parents=False):
     """
     Stream from/to local filesystem, transparently (de)compressing gzip and bz2
     files if necessary.
 
     """
+    if parents:
+        from ntpath import split
```

Comment:
http://www.it-recipes.com/articles/blog/161 gives a good explanation on why I used `ntpath` instead of `os.path`. Quoting:

> Linux filenames may contain backslashes. So on linux, r'a/b\c' always refers to the file b\c in the a folder, while on Windows, it always refers to the c file in the b subfolder of the a folder. So when both forward and backward slashes are used in a path, you need to know the associated platform to be able to interpret it correctly. In practice it's usually safe to assume it's a windows path since backslashes are seldom used in Linux filenames, but keep this in mind when you code so you don't create accidental security holes.